docs/Account/methods.md

```diff
@@ -0,0 +1,284 @@
+
+      ```ts
+      type NonceOptions = {
+        nonceKey?: number;
+        nonceOverride?: number;
+      };
+      // nonceOptions usage
+      let i = 0;
+      const userOp = await smartAccount.buildUserOp([tx1], {
+        nonceOptions: { nonceKey: i++ },
+      });
+      ```
+      Nonce can be initialised at any arbitrary number and incremented as one builds transactions. The nonceKey will create a batch or space in which the nonce can safely increment without colliding with other transactions. The nonceOverride will directly override the nonce and should only be if you know the order in which you are sending the userOps.
+
+    
+    
+  5. forceEncodeForBatch (`boolean`): 
+
+  6. paymasterServiceData (`SponsorUserOperationDto`): The `paymasterServiceData` includes details about the kind of sponsorship and payment token. It contains information about the paymaster service, which is used to calculate the `paymasterAndData` field in the user operation.
+
+      ```ts
+      type SponsorUserOperationDto = {
+        mode: PaymasterMode;
+        calculateGasLimits?: boolean; // Add info
+        expiryDuration?: number;
+        webhookData?: {
+            [key: string]: any;
+        };
+        smartAccountInfo?: SmartAccountData;
+        feeTokenAddress?: string;
+      }
+      ```
+      It also contains optional fields such as `webhookData` and `smartAccountInfo`.
+  
+
+**Returns**
+- partialUserOp (`Promise<Partial<UserOperation>>`): A Promise resolving to `partialUserOp` which can be further sent for execution⁠.
+
+### senduserOp( )
+
+This method is used to submit a User Operation object to the User Operation pool of the client. It signs the UserOperation using an externally owned account (EOA) key and submits it to the bundler for on-chain processing. 
+
+**Usage**
+
+```ts
+const userOpResponse = await smartAccount.sendUserOp(userOp);
+
+type UserOpResponse = {
+  userOpHash: string;
+  wait(_confirmations?: number): Promise<UserOpReceipt>;
+  waitForTxHash(): Promise<UserOpStatus>;
+};
+
+const { receipt } = await userOpResponse.wait(1);
+
+type UserOpReceipt = {
+  userOpHash: string;
+  entryPoint: string;
+  sender: string;
+  nonce: number;
+  paymaster: string;
+  actualGasCost: BigNumber;
+  actualGasUsed: BigNumber;
+  success: boolean;
+  reason: string;
+  logs: Array<ethers.providers.Log>;
+  receipt: ethers.providers.TransactionReceipt;
+};
+```
+
+**Parameters**
+- userOp (`Partial<UserOperation>`, required): The `userOp` object includes essential fields like `sender`, `nonce`, `callData`, `gas` related properties.
+
+- params (`SendUserOpParams`): The `SendUserOpParams` object can contain fields such as `sessionID`, `sessionSigner`, `sessionValidationModule`, `additionalSessionData`, `batchSessionParams`, and `simulationType`. These parameters are used to customize the behavior of the `sendUserOp` method and are optional. 
```

Comment:
are used when your active validation module is complex and requires additional information for signature generation.